### DISCUSSION TITLE
Difference Between Local Nominatim and OpenStreetMap Website results

### DISCUSSION THREAD

**Discussion by mbeneventi** [osm-search/Nominatim#2839] [Q&A] [answered]
have installed nominatim 4.1.0 (tokenizer= ICU) via following instructions on [nominatim documentation](https://nominatim.org/release-docs/latest/admin/Installation/), added wikipedia data during the installation, and imported an updated pbf file from [geofabrik.de](https://download.geofabrik.de/europe/italy/isole.html).

All works but when I sent some kind of request (e.g. Cagliari via Roma) the answer I get from Nominatim Website (https://nominatim.openstreetmap.org/) and my local installation are very different. The right results are on nominatim website of course.

The problems seems to be with the search candidate algorithm or the attribuition/calc of AdressImportance parameter. The very strange thing is that I get these wrong results only for some requests.

There is any particular parameter to set or anything else to verify?

I hope it is clear for you and even small advice or comment would be very helpful for me

Thanks Michele

**Comment by mtmail** [comment, marked as answer]
Thanks for providing the details of your installation (versions, wikipedia imported, which region imported). Can you post, ideally the full JSON output what queries seem wrong and what results you got? It's a bit of guessing but maybe we can spot missing information. 
You can also add `&debug=1` to the URL to compare two Nominatim instances. That will print HTML with extra information. Example output https://nominatim.openstreetmap.org/search.php?q=roma&format=jsonv2&debug=1

**Reply by mbeneventi** [reply]
Hi mtmail ,
the query is the following:
http://localhost:8088/search.php?q=cagliari+via+roma&format=jsonv2&debug=1

Attached the output files

[search_local.zip](https://github.com/osm-search/Nominatim/files/9751701/search_local.zip)
[search_nominatim_OSM.zip](https://github.com/osm-search/Nominatim/files/9751703/search_nominatim_OSM.zip)

**Reply by mtmail** [reply]
So the best results is the 'Via Roma' in the city of Cagliari (https://www.openstreetmap.org/relation/39837) in the Marina neighboorhood and not any of the other 'Via Roma' in the county by the same name (https://www.openstreetmap.org/relation/276369).

There's one change https://github.com/osm-search/Nominatim/commit/e0c184e097 that got added after the 4.1.0 stable release related to https://github.com/osm-search/Nominatim/issues/882 Can you check if the 'limit' paramter makes a difference? E.g. setting it to 20 or 50?

If you edit the file (`lib-php/Geocode.php`) then remember to restart your webserver and/or php-fpm. Or try the query on the command-line: `nominatim search ...`

**Reply by mbeneventi** [reply]
Hi,
I edited the file Geocode.php with 'max' and tried both values

 
$this->iLimit = $iLimit + max($iLimit, 50)



 $this->iLimit = $iLimit + max($iLimit, 20)

and restart the webserver but  ​​unfortunately the results are wrong 


[search_20.zip](https://github.com/osm-search/Nominatim/files/9762749/search_20.zip)
[search_50.zip](https://github.com/osm-search/Nominatim/files/9762750/search_50.zip)

**Reply by mbeneventi** [reply]
Hi,
another consideration:
Seems that in my local installation the algorithm is discarding results with the highest `addressimportance` parameter value by putting them in the queue. 
Is there any configuration I'm setting wrong?

thanks
Michele

**Reply by lonvia** [reply]
There are about 160 results for 'Via Roma Cagliari'. If the right one is among the results is random, even with a limit of 50 results. #2268 will fix that eventually.

**Reply by mbeneventi** [reply]
I understood your reasoning but I wonder why the online version works correctly. I think there is something different in my local installation

**Reply by lonvia** [reply]
Postgresql happens to have saved the OSM objects in a different order in the database and therefore returns different ones. It will be different on every installation.

**Reply by mbeneventi** [reply]
I found an acceptable solution by editing the Geocode.php file line as follows:

`$this->iLimit = $iLimit + max($iLimit, 150)`

the result is not exactly the same as that of the online nominatim but it works fine for my goals

thank you
